pdns/recursordist/ws-recursor.cc

```diff
@@ -384,48 +389,46 @@ static void apiServerZones(HttpRequest* req, HttpResponse* resp)
   resp->setJsonBody(doc);
 }
 
-static void apiServerZoneDetail(HttpRequest* req, HttpResponse* resp)
-{
-  DNSName zonename = apiZoneIdToName(req->parameters["id"]);
-
-  auto iter = SyncRes::t_sstorage.domainmap->find(zonename);
-  if (iter == SyncRes::t_sstorage.domainmap->end()) {
-    throw ApiException("Could not find domain '" + zonename.toLogString() + "'");
+#define findZoneById(req)                                                         \
```

Comment:
In the case of rec, this can  be a function returning a `DNSName`. The callers then become:
```
auto zonename = findZoneById(req);
```